src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-share-form/smb-share-form.component.html

```diff
@@ -0,0 +1,229 @@
+<div cdsCol
+     [columnNumbers]="{ md: 4 }">
+  <form name="smbShareForm"
+        #formDir="ngForm"
+        [formGroup]="smbShareForm"
+        novalidate>
+    <div i18n="form title"
+         class="form-header">
+      {{ action | titlecase }} {{ resource | upperFirst }}
+    </div>
+
+    <!-- Share Id -->
+    <div class="form-item">
+      <cds-text-label
+        for="share_id"
+        i18n
+        helperText="Unique share identifier"
+        i18n-helperText
+        cdRequiredField="Share Name"
+        [invalid]="smbShareForm.controls.share_id.invalid && smbShareForm.controls.share_id.dirty"
+        [invalidText]="shareError"
+        >Share Name
+        <input
+          cdsText
+          type="text"
+          id="share_id"
+          formControlName="share_id"
+          [invalid]="smbShareForm.controls.share_id.invalid && smbShareForm.controls.share_id.dirty"
+        />
+      </cds-text-label>
+      <ng-template #shareError>
+        <span
+          class="invalid-feedback"
+          *ngIf="smbShareForm.showError('share_id', formDir, 'required')"
+          i18n
+          >This field is required.</span
+        >
+      </ng-template>
+    </div>
+
+    <!-- Volume -->
+    <div class="form-item">
+      <cds-select
+        formControlName="volume"
+        for="volume"
+        label="Volume"
+        cdRequiredField="Volume"
+        id="volume"
+        (change)="volumeChangeHandler()"
+        [invalid]="smbShareForm.controls.volume.invalid && smbShareForm.controls.volume.dirty"
+        [invalidText]="volumeError"
+      >
+        <option *ngIf="allFsNames?.length === 0"
+                value=""
+                i18n>
+          -- No filesystem available --
+        </option>
+        <option *ngIf="allFsNames !== null && allFsNames?.length > 0"
+                value=""
+                i18n>
+          -- Select the filesystem --
+        </option>
+        <option *ngFor="let filesystem of allFsNames"
+                [value]="filesystem.name"
+                i18n>
+          {{ filesystem.name }}
+        </option>
+      </cds-select>
+      <ng-template #volumeError>
+        <span
+          class="invalid-feedback"
+          *ngIf="smbShareForm.showError('volume', formDir, 'required')"
+          i18n
+          >This field is required.</span
+        >
+      </ng-template>
+    </div>
+
+    <div class="form-item"
+         *ngIf="smbShareForm.getValue('volume')">
+      <cds-select
+        formControlName="subvolume_group"
+        for="subvolume_group"
```

Comment:
:) `id` is used as `for` @dnyanee1997 
```suggestion
```

https://github.com/carbon-design-system/carbon-components-angular/blob/cd447b6cf13310f650036572419c15f1b81e3470/src/select/select.component.ts#L67C1-L68C1